src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -2138,6 +2142,39 @@ public void LeaderActions()
             ShutdownSelfWhenDown();
         }
 
+        private void MoveJoiningToWeaklyUp()
+        {
+            var localGossip = _latestGossip;
+            var localMembers = localGossip.Members;
+            var enoughMembers = IsMinNrOfMembersFulfilled();
+
+            bool IsJoiningToWeaklyUp(Member m) => m.Status == MemberStatus.Joining
+                                                  && enoughMembers
+                                                  && _latestGossip.ReachabilityExcludingDownedObservers.Value.IsReachable(m.UniqueAddress);
+
+            var changedMembers = localMembers
+                .Where(IsJoiningToWeaklyUp)
+                .Select(m => m.Copy(MemberStatus.WeaklyUp))
+                .ToImmutableSortedSet();
+
+            if (!changedMembers.IsEmpty)
+            {
+                // replace changed members
+                var newMembers = changedMembers.Union(localMembers);
```

Comment:
Son of a bitch... I totally missed this when I originally reviewed the PR. It's going to great the same problems here that happened in https://github.com/akkadotnet/akka.net/issues/2584. This is a bug. I'll PR it per https://github.com/akkadotnet/akka.net/issues/3274